packages/node/src/indexer/blockDispatcher/worker-block-dispatcher.service.ts

```diff
@@ -200,11 +202,14 @@ export class WorkerBlockDispatcherService
     // Used to compare before and after as a way to check if queue was flushed
     const bufferedHeight = this.latestBufferedHeight;
 
+    // get SpecVersion from main runtime service
+    const { blockSpecVersion, syncedDictionary } =
```

Comment:
Because we are calling `enqueueBlock` from a promise.all will this lead for fetching the same spec version more than once?

Reply:
possible when runtime service is in the process of syncing with dictionary